src/shared/lib/URLWrapper.ts

```diff
@@ -0,0 +1,51 @@
+import {autorun, computed, extendObservable, intercept, IObservableObject, observable} from "mobx";
+import ExtendedRouterStore from "./ExtendedRouterStore";
+
+export type Property<T> = {
+    name: keyof T,
+    persisted: boolean
+};
+
+export default class URLWrapper<QueryParamsType> {
+    public query:Partial<QueryParamsType>;
+
+    constructor(
+        protected routing:ExtendedRouterStore,
+        protected properties:Property<QueryParamsType>[]
+    ) {
+        const initValues:Partial<QueryParamsType> = {};
+        for (const property of properties) {
+            initValues[property.name] = undefined;
+        }
+        this.query = observable<Partial<QueryParamsType>>(initValues);
+
+        intercept(this.query, change=>{
```

Comment:
interesting.  didn't realize you could set one for all properties.  and the callback fires for each individual property/

Reply:
yeah!